### PR TITLE
Feature/load misc items

### PR DESCRIPTION
**Pull request by Ripolak**
Allow rendering icons of misc items from the misc.txt file for viewing in inventory.

### CODE REVIEW

**Comment by gravestench** [comment]
Aren't these already loaded by item_common ?

**Comment by Ripolak** [comment]
> Aren't these already loaded by item_common ?

Not for rendering on screen from what I've seen, but as logical structs. I might be missing something though, do correct me if I'm wrong.

**Comment by gravestench** [comment]
I made a stupid assumption instead of reading into your commit, apologies.

File: d2core/d2inventory/inventory_item_misc.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package d2inventory
+
+import (
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2data/d2datadict"
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2enum"
+	"log"
+)
+
+type InventoryItemMisc struct {
```
Comment (essial) [inline d2core/d2inventory/inventory_item_misc.go:9]: Could we make this more generic and simply inherit properties? InventoryItemArmor and Weapon both use these fields as well, so it'd make sense to make a generic version and just extend for those.

Comment (Ripolak) [reply d2core/d2inventory/inventory_item_misc.go:9]: Could be, those 3 structs actually already implement an `InventoryItem` interface (d2core/inventory/inventory_item.go), how'd you suggest we make it more generic?

Comment (essial) [reply d2core/d2inventory/inventory_item_misc.go:9]: Gotcha, I'll accept for now, we can clean it up later. Thanks again!

**Pull request merged**